torchgeo/datasets/caffe.py

```diff
@@ -0,0 +1,293 @@
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new instance of CaFFe dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert split in self.valid_splits, f'split must be one of {self.valid_splits}'
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        self.fpaths = glob.glob(
+            os.path.join(
+                self.root,
+                self.zipfilename.replace('.zip', ''),
+                self.mask_dirs[1],
+                self.split,
+                '*.png',
+            )
+        )
+
+        self.ordinal_map_zones = torch.zeros(
+            max(self.px_class_values_zones.keys()) + 1, dtype=torch.long
+        )
+        for ordinal, px_class in enumerate(self.px_class_values_zones.keys()):
+            self.ordinal_map_zones[px_class] = ordinal
+
+        self.ordinal_map_fronts = torch.zeros(
+            max(self.px_class_values_fronts.keys()) + 1, dtype=torch.long
+        )
+        for ordinal, px_class in enumerate(self.px_class_values_fronts.keys()):
+            self.ordinal_map_fronts[px_class] = ordinal
+
+    def __len__(self) -> int:
+        """Return the number of images in the dataset."""
+        return len(self.fpaths)
+
+    def __getitem__(self, idx: int) -> dict[str, Tensor]:
+        """Return the image and mask at the given index.
+
+        Args:
+            idx: index of the image and mask to return
+
+        Returns:
+            dict: a dict containing the image and mask
+        """
+        zones_filename = os.path.basename(self.fpaths[idx])
+        img_filename = zones_filename.replace('_zones_', '_')
+        front_filename = zones_filename.replace('_zones_', '_front_')
+
+        def read_tensor(path: str) -> Tensor:
+            return torch.from_numpy(np.array(Image.open(path)))
+
+        img = (
+            read_tensor(
+                os.path.join(
+                    self.root, self.data_dir, self.image_dir, self.split, img_filename
+                )
+            )
+            .unsqueeze(0)
+            .float()
+        )
```

Comment:
These long lines could be condensed quite a bit using something like:
```suggestion
        path = os.path.join(self.root, self.data_dir, self.image_dir, self.split, img_filename)
        img = read_tensor(path).unsqueeze(0).float()
```